Sources/OpenId4VP/OpenId4VP.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+
+public class OpenId4VP {
+    let traceabilityId: String
+    let networkManager: NetworkManaging
+    var authorizationRequest: AuthorizationRequest?
+    static var presentationDefinitionId: String?
+    static var responseUri: String?
+    
+    init(traceabilityId: String, networkManager: NetworkManaging? = nil) {
+        self.traceabilityId = traceabilityId
+        self.networkManager = networkManager ?? NetworkManager.shared
+    }
+    
+    public static func setPresentationDefinitionId(_ id: String) {
+        OpenId4VP.presentationDefinitionId = id
+    }
+    
+    public static func setResponseUri(_ responseUri: String){
+        OpenId4VP.responseUri = responseUri
+    }
+    
+    public func authenticateVerifier(encodedAuthorizationRequest: String, trustedVerifierJSON: [[Verifier]]) async throws -> AuthenticationResponse {
+        
+        Logger.setLogTag(className:String(describing: type(of: self)), traceabilityId: traceabilityId)
+        
+        do {
+            authorizationRequest =  try AuthorizationRequest.getAuthorizationRequest(encodedAuthorizationRequest: encodedAuthorizationRequest)
+            
+            return try AuthenticationResponse.getAuthenticationResponse(authorizationRequest!, trustedVerifierJSON)
+            
+        } catch(let exception) {
+            await sendErrorToResponseUri(error: exception, uri: OpenId4VP.responseUri!)
+            throw exception
+        }
+    }
+    
+    public func constructVerifiablePresentationToken(credentialsMap: [String: [String]]) async throws ->  String? {
+        
+        return try AuthorizationResponse.constructVpForSigning(credentialsMap)
+    }
+    
+    public func shareVerifiablePresentation(vpResponseMetadata: VPResponseMetadata) async throws -> String? {
+        
+        do {
+            return try await AuthorizationResponse.shareVp(vpResponseMetadata: vpResponseMetadata,nonce: authorizationRequest!.nonce, responseUri: authorizationRequest!.responseUri,presentationDefinitionId: OpenId4VP.presentationDefinitionId!, networkManager: networkManager)
```

Comment:
why do we need to take reference as `OpenId4VP.presentationDefinitionId` thought it's in same class

Reply:
Since `shareVerifiablePresentation` is an instance method and presentationDefinitionId is a static member of the class, xcode is not allowing to access it using the instance type of the class.